proto/dataplane/dataplane.proto

```diff
@@ -0,0 +1,133 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package lemming.dataplane;
+
+option go_package = "github.com/openconfig/lemming/proto/dataplane";
+
+message UpdatePortRequest {
+  string name = 1;
+  string hwaddr = 2;
+  repeated IPNetwork ipv4 = 3;
+  repeated IPNetwork ipv6 = 4;
+}
+
+message UpdatePortResponse {}
+
+enum PortState {
+  PORTSTATE_INVALID = 0;
+  PORTSTATE_DOWN = 1;
+  PORTSTATE_UP = 2;
+}
+
+message SetPortStateRequest {
+  string name = 1;
+  PortState admin_state = 2;
+}
+
+message SetPortStateResponse {}
+
+message SubscribePortStateRequest {}
+
+message IPNetwork {
+  string ip = 1;
+  uint64 prefix_len = 3;
+}
+
+message SubscribePortStateResponse {
+  string name = 1;
+  PortState admin_state = 2;
+  repeated IPNetwork network = 3;
+}
+
+message RouteDst {
+  string port = 1;
+  string ip = 2;
+  uint64 weight = 3;
+}
+
+message InsertRouteRequest {
+  uint64 vrf = 1;
+  string prefix = 2;
+  repeated RouteDst dst = 3;
+}
+
+message InsertRouteResponse {}
+
+message DeleteRouteRequest {
+  string prefix = 1;
+  uint64 vrf = 2;
+}
+
+message DeleteRouteResponse {}
+
+message AddNeighborRequest {
+  string ip = 1;
+  string hwaddr = 2;
+  bool transient = 3;
```

Comment:
Does transient here mean that this is an ARP entry that should time out? If so, did we consider maybe just having this be implemented outside of the dataplane? i.e., if an ARP entry is meant to timeout then the daemon that installed it should go remove it when it wants it to time out?

Reply:
yeah, that was the intention. I'm in favor that being handled outside the dataplane